test/parallel/test-vm-basic.js

```diff
@@ -128,3 +128,30 @@ const vm = require('vm');
              'Received type object'
   });
 });
+
+// vm.compileFunction
```

Comment:
This needs tests for failed validation of each option.

Reply:
Done.